### PR TITLE
Frontend integration

### PR DESCRIPTION
**Pull request by aaqibqadeer**

### CODE REVIEW

File: packages/shopping-cart/src/pages/Cart.jsx [hunk 1]
```diff
@@ -14,26 +14,54 @@ export const Cart = () => {
...
     const updatedCart = [...cart];
-    const productIndex = updatedCart.findIndex((item) => item.id === id);
+    const productIndex = updatedCart.findIndex((item) => item._id === _id);
     updatedCart[productIndex].quantity = quantity;
     updateCart(updatedCart);
   };
 
-  const handleRemove = (id) => {
-    const updatedCart = cart.filter((item) => item.id !== id);
+  const handleRemove = (_id) => {
```
Comment (Lutif) [inline packages/shopping-cart/src/pages/Cart.jsx]: ` const handleRemove = (id) => {`

File: packages/shopping-cart/src/pages/Cart.jsx [hunk 2]
```diff
@@ -47,42 +75,28 @@ export const Cart = () => {
...
       if (cart.length > 0) {
         cart.forEach((item) => {
-          const product = getProductInfo(item.id);
+          const product = getProductInfo(item._id);
           price += product.price * item.quantity;
         });
         history.push({
           pathname: "/checkout",
-          state: { subtotal: price },
+          state: { subtotal: price, cart: cart },
```
Comment (Lutif) [inline packages/shopping-cart/src/pages/Cart.jsx]: `  state: { subtotal: price, cart },`

File: packages/shopping-cart/src/pages/Checkout.jsx [hunk 3]
```diff
@@ -21,10 +23,35 @@ export const Checkout = () => {
   const [address, setAddress] = useState("");
   const [number, setNumber] = useState("");
   const [paymentMethod, setPaymentMethod] = useState("COD");
+  const { updateLoading } = useContext(LoadingContext);
```
Comment (Lutif) [inline packages/shopping-cart/src/pages/Checkout.jsx]: does loading really need to be in context ?

File: packages/shopping-cart/src/pages/Checkout.jsx [hunk 4]
```diff
@@ -21,10 +23,35 @@ export const Checkout = () => {
...
+    const newOrder = {
+      userId: userId,
+      productList: formatProductList(location.state.cart),
+      checkoutDetails: {
+        fullname: name,
+        country: country,
+        address: address,
+        number: number,
+        paymentMethod: paymentMethod,
+      },
```
Comment (Lutif) [inline packages/shopping-cart/src/pages/Checkout.jsx:49]: `userId,
      productList: formatProductList(location.state.cart),
      checkoutDetails: {
        fullname: name,
        country,
        address,
        number,
        paymentMethod,`

File: packages/shopping-cart/src/utils/api/useProductsHook.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import { useCallback, useState } from "react";
+import { api } from "./api";
+
+export const useProductsHook = () => {
+  const [products, setProducts] = useState({});
+  const [res, setRes] = useState({
+    success: false,
+    loading: false,
+    status: null,
```
Comment (Lutif) [inline packages/shopping-cart/src/utils/api/useProductsHook.jsx:8]: why not create a 4th field `products` ?

File: packages/shopping-cart/src/utils/api/useLoginHook.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  const login = useCallback(async (payload) => {
+    setRes({
+      success: false,
+      loading: true,
+      status: null,
+    });
+
+    try {
+      const response = await api.post("/user/login", payload);
+      localStorage.setItem("userId", response.data._id);
```
Comment (Lutif) [inline packages/shopping-cart/src/utils/api/useLoginHook.jsx:30]: where does user data go?

File: packages/shopping-cart/src/pages/Checkout.jsx (hunk 4 above)
Comment (aaqibqadeer) [reply packages/shopping-cart/src/pages/Checkout.jsx:49]: I don't know why this and the issues below aren't outdated. 
I have made changes in these components as well.
https://github.com/aaqibqadeer/shopping-cart/blob/frontend-integration/packages/shopping-cart/src/pages/Checkout.jsx

**Pull request merged**